boards/DIM/Src/App/App_SevenSegDisplays.c

```diff
@@ -0,0 +1,55 @@
+#include <stdlib.h>
+
+#include "App_SevenSegDisplays.h"
+#include "App_SharedAssert.h"
+
+struct SevenSegDisplays
+{
+    float (*get_state_of_charge)(void);
+    void (*set_left_hex_digit)(uint8_t);
```

Comment:
Previously, I created struct for **individual** 7-seg displays. And instead of `void (*set_left_hex_digit)(uint8_t);` I had `struct SevenSegDisplay* left_display` but later I realized `struct SevenSegDisplay` only has one method function.

Considering how much boilerplate code I'd have to add for `struct SevenSegDisplay`, I decided to just replace that with a simple setter function pointer here.

Reply:
Depends if you think this will ever change or not. Personally I'd be inclined to have a struct per-display, but I'm fine with this as well.

Reply:
Hmm it's one extra level of indrection, i.e.
```
SevenSegDisplays
  SevenSegDisplay
    Io_SeLeftDigit
```
v.s. 
```
SevenSegDisplays
    Io_SeLeftDigit
```
But I guess the former is more logically sound? I'll just sit on this a little more and maybe revert the changes. But yeah I'm fine with either option too atm.

Reply:
Yeah, I prefer the first option here.

Reply:
Added `SevenSegDisplay` back in. You can see in `main.c` that I had to create left, middle, and right displays individually.